### PR TITLE
spruce up features (lineage, catalog, diff, alerts)

### PR DESCRIPTION
**Pull request by leoebfolsom**
I found the documentation for Lineage lacking--basically, I couldn't find Lineage in the Datafold app. It wasn't obvious to me that you had to go to Catalog and click on a table to see its Lineage.

I've added a few points that I think will help someone with that. 

It's a bit of an awkward fit because the Features pages (lineage, catalog, diff, alerts) are framed more as "here are the great benefits" more than "how-to docs." I think longer term, these features deserve their own explainers in the broader docs section; even though they're (by design) not technically complex to use, the user still needs some basic instructions that don't really belong in the Features page.

But, I think this is worth merging as a quick fix / incremental improvement until we create a more holistic solution.

### CODE REVIEW

File: docs/features/catalog-lineage.md [hunk 1]
```diff
@@ -4,13 +4,19 @@ title: Catalog + Column-level Lineage
...
 
-![](../../static/img/lineage_detail.png)
\ No newline at end of file
+No additional developer resources are needed to unlock this capability. Simply connect your data warehouse and explore your lineage graph. 
+
+To view the column-level Lineage of any table in your warehouse:
+
+* Navigate to Catalog.
+* Click on any table in the Catalog.
+* Click on Lineage.
```
Comment (dlawin) [inline docs/features/catalog-lineage.md:22]: There would be an additional step to get to column level lineage here: expanding the columns caret

File: docs/overview.mdx [hunk 2]
```diff
@@ -28,7 +28,7 @@ import TabItem from '@theme/TabItem';
 
 <table >
  <tr>
-    <td>Get the row-level impact of changes you’re making to your dbt models within your workflow – all without needing to write any SQL tests.</td>
+    <td>Get the row-level impact of changes you're making to your dbt models within your workflow – all without needing to write any SQL tests.</td>
```
Comment (dlawin) [inline docs/overview.mdx:31]:
```suggestion
    <td>Get the row-level impact of changes made to dbt models within your workflow – all without needing to write any SQL tests.</td>
```

Comment (dlawin) [reply docs/overview.mdx:31]: this flows a little better in my head

File: docs/features/data_diff.md [hunk 3]
```diff
@@ -3,11 +3,11 @@ sidebar_position: 1
 title: Data Diff
 ---
 # Data Diff
-Data Diff enables teams to become proactive with their data testing and easily allows you to validate migrations before breaking changes have a chance to make it to production. 
+Data Diff enables teams to be proactive with their data testing. With Data Diff, you can easily validate migrations before breaking changes make it to production.
```
[the remark this answers is not in the document]
Comment (leoebfolsom) [reply docs/features/data_diff.md:6]: I agree, I've updated it to "you can easily validate that updates to your data pipelines won't cause breaking changes in production," but we should revisit all of this language.

**Review by williebsweet — APPROVED** [review]

**Pull request merged**